src/FunderMaps.Infrastructure/Storage/SpacesBlobStorageService.cs

```diff
@@ -189,26 +194,109 @@ public async Task StoreDirectoryAsync(string directoryName, string directoryPath
+        /// <returns>See <see cref="ValueTask"/>.</returns>
+        public async Task RemoveDirectoryAsync(string directoryPath)
+        {
+            try
+            {
+                // NOTE: This call returns max. up to 1000 records by design
+                //       In order to obtain every record that matches our query, 
+                //       we have to repeat our request execution in a loop - using continuation tokens -- until no longer truncated.
+                //       See https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/S3/MS3ListObjectsV2AsyncListObjectsV2RequestCancellationToken.html
+                ListObjectsV2Request request = new()
+                {
+                    BucketName = _options.BlobStorageName,
+                    Prefix = directoryPath,
+                    MaxKeys = MaxKeys
+                };
+
+                List<Task> tasklist = new();
+
+                ListObjectsV2Response response;
+                do
+                {
+                    response = await client.ListObjectsV2Async(request);
+
+                    if (response.S3Objects.Count <= 0)
                     {
-                        using Stream stream = File.OpenRead(file);
-                        await StoreFileAsync(directoryName, Path.Combine(subdir, Path.GetFileName(file)), storageObject.ContentType, stream, storageObject);
+                        continue;
                     }
-                    foreach (var file in Directory.GetDirectories(path))
+
+                    Task deleteTask = client.DeleteObjectsAsync(new()
                     {
-                        await DirectorySearch(file, Path.Combine(subdir, Path.GetFileName(file)));
-                    }
-                }
+                        BucketName = _options.BlobStorageName,
+                        Objects = response.S3Objects.Select(x => new KeyVersion()
+                        {
+                            Key = x.Key
+                        }).ToList()
+                    });
+
+                    tasklist.Add(deleteTask);
+
+                    _ = Task.Run(async () =>
+                    {
+                        _logger.LogTrace($"(Task {Task.CurrentId}): Starting remote deletion of {response.S3Objects.Count} objects. ");
 
-                await DirectorySearch(directoryPath);
+                        await deleteTask;
+
+                        _logger.LogTrace($"(Task {Task.CurrentId}): Completed.");
+                    });
+
+                    request.ContinuationToken = response.NextContinuationToken;
+                } while (response.IsTruncated);
+
+                await Task.WhenAll(tasklist.ToArray());
             }
             catch (AmazonS3Exception e)
             {
-                _logger.LogError(e, "Could not store file to Spaces using S3");
+                _logger.LogTrace(e, "Could not delete directory on Spaces using S3");
```

Comment:
There is no need to print the stacktrace here. Remove the `e` from logTrace. This is true in all of the solution